### PR TITLE
Clear all map elements before generating a new map, fixes #1762

### LINKED ISSUE #1762: Scenario Editor Terrain Generation & Park Entrances
If I place one or more park entrances, and generate some terrain, terrain generation removes all of the park entrances as expected, but the editor thinks the entrances are still there and won't let you finish because it's impossible to place a path to them. If you place all four entrances, it's impossible to place any entrance after terrain generation.

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by IntelOrca** [comment]
This works, but there is probably still other initialisation that is required such as resetting of peep spawn positions and guaranteed reset of park entrance positions.

**Comment by Gymnasiast** [comment]
sub_68AE2A should handle that unless I'm very mistaken, since the things you mentioned should also came up when decreasing the map size (for which sub_68AE2A is also used).

**Comment by IntelOrca** [comment]
sub_68AE2A only handles that if there is a corresponding map element. But it is not a full reset map initialisation function.

File: src/world/map.c [hunk 1]
```diff
@@ -3954,3 +3954,14 @@ bool map_surface_is_blocked(sint16 x, sint16 y){
 	return false;
 }
 
+/* Clears all map elements, to be used before generating a new map */
+void map_clear_all_elements()
+{
+	int mapMaxXY = RCT2_GLOBAL(RCT2_ADDRESS_MAP_MAX_XY, uint16);
```
Comment (AaronVanGeffen) [inline src/world/map.c]: This variable is left unused, isn't it? Should it be used instead of the hardcoded `256`?

Comment (Gymnasiast) [inline src/world/map.c]: Possibly. I would have to check.

**Pull request merged**